test/pgn_test.dart

```diff
@@ -0,0 +1,209 @@
+          eval: PgnEvaluation.pawns(pawns: -0.42),
+        ));
+
+    expect(
+        PgnComment.fromPgn('prefix [%emt 1:02:03.4]'),
+        const PgnComment(
+          text: 'prefix',
+          emt: 3723.4,
+        ));
+
+    expect(
+        PgnComment.fromPgn('[%csl Ya1][%cal Ra1a1,Be1e2]commentary [%csl Gh8]'),
+        const PgnComment(text: 'commentary', shapes: [
+          PgnCommentShape(color: CommentShapeColor.yellow, from: 0, to: 0),
+          PgnCommentShape(color: CommentShapeColor.red, from: 0, to: 0),
+          PgnCommentShape(color: CommentShapeColor.blue, from: 4, to: 12),
+          PgnCommentShape(color: CommentShapeColor.green, from: 63, to: 63)
+        ]));
+
+    expect(
+        PgnComment.fromPgn('prefix [%eval .99,23]'),
+        const PgnComment(
+            text: 'prefix', eval: PgnEvaluation.pawns(pawns: 0.99, depth: 23)));
+
+    expect(
+        PgnComment.fromPgn('[%eval #-3] suffix'),
+        const PgnComment(
+          text: 'suffix',
+          eval: PgnEvaluation.mate(mate: -3),
+        ));
+
+    expect(
+        PgnComment.fromPgn('[%csl Ga1]foo'),
+        const PgnComment(text: 'foo', shapes: [
+          PgnCommentShape(color: CommentShapeColor.green, from: 0, to: 0)
+        ]));
+
+    expect(
+        PgnComment.fromPgn(
+                'foo [%bar] [%csl Ga1] [%cal Ra1h1,Gb1b8] [%clk 3:25:45]')
+            .text,
+        'foo [%bar]');
+  });
+
+  test('make comment', () {
+    expect(
+        const PgnComment(
+            text: 'text',
+            emt: 3723.4,
+            eval: PgnEvaluation.pawns(pawns: 10),
+            clock: 1,
+            shapes: [
+              PgnCommentShape(color: CommentShapeColor.yellow, from: 0, to: 0),
+              PgnCommentShape(color: CommentShapeColor.red, from: 0, to: 1),
+              PgnCommentShape(color: CommentShapeColor.red, from: 0, to: 2)
+            ]).makeComment(),
+        'text [%csl Ya1] [%cal Ra1b1,Ra1c1] [%eval 10.00] [%emt 1:02:03.4] [%clk 0:00:01]');
+
+    expect(
+        const PgnComment(eval: PgnEvaluation.mate(mate: -4, depth: 5))
+            .makeComment(),
+        '[%eval #-4,5]');
+  });
+
+  test('roundtrip comment', () {
+    final comments = [
+      '[%csl[%eval 0.2] Ga1]',
+      '[%c[%csl [%csl Ga1[%csl Ga1][%[%csl Ga1][%cal[%csl Ga1]Ra1]',
+      '[%csl Ga1][%cal Ra1h1,Gb1b8] foo [%clk 3:ê5: [%eval 450752] [%evaÿTæ<92>ÿÿ^?,7]',
+    ];
+
+    for (final str in comments) {
+      final comment = PgnComment.fromPgn(str);
+      final roundTripped = PgnComment.fromPgn(comment.makeComment());
+      expect(comment, roundTripped);
+    }
+  });
+
+  test('transform pgn', () {
+    final game = PgnGame.parsePgn('1. a4 ( 1. b4 b5 -- ) 1... a5');
+    final res = game.moves.transform<TransformResult, Position>(
+      Chess.initial,
+      // update test and use parseSan to update position.
+      (pos, data, idx) => TransformResult(fen: pos.fen, san: data.san),
+    );
+
+    expect(res.children[0].data.fen,
```

Comment:
this test has little value... but we miss the `parseSAN` ; it would be nice to actually add the `parseSAN` in this PR as well

Reply:
Yes will add the proper test when `parsePgn` is completed